core/src/main/java/org/apache/iceberg/util/SortOrderUtil.java

```diff
@@ -37,16 +37,14 @@ private SortOrderUtil() {
   }
 
   public static SortOrder buildSortOrder(Table table) {
-    return buildSortOrder(table.spec(), table.sortOrder());
+    return buildSortOrder(table.schema(), table.spec(), table.sortOrder());
   }
 
-  public static SortOrder buildSortOrder(PartitionSpec spec, SortOrder sortOrder) {
+  public static SortOrder buildSortOrder(Schema schema, PartitionSpec spec, SortOrder sortOrder) {
```

Comment:
Is this used anywhere? Maybe we should make it private instead of just changing it?

Reply:
I think the original API you guys had internally had this syntax. In OSS, we have access to `Table` wherever this is used. Internally, we temporarily use this version but I have no issues making it private in OSS. Let me know if you use this version, @rdblue.

Reply:
I don't think we use it. We should be able to pass the table, I think.

Reply:
Sounds good. Let me update it.

Reply:
Fixed.